x-pack/test/functional/apps/maps/import_geojson/add_layer_import_panel.js

```diff
@@ -0,0 +1,159 @@
+          path.join(__dirname, FILE_LOAD_DIR, secondLoadFileName)
+        );
+        await PageObjects.maps.waitForLayersToLoad();
+        // Check second file is loaded in file picker
+        const filePickerLoadedFile = await PageObjects.maps
+          .hasFilePickerLoadedFile(secondLoadFileName);
+        expect(filePickerLoadedFile).to.be(true);
+        // Check that no file is loaded in layer preview
+        const layerLoadedInToc = await PageObjects.maps.doesLayerExist(IMPORT_FILE_PREVIEW_NAME);
+        expect(layerLoadedInToc).to.be(false);
+      });
+
+    it('should supply the correct index type(s) (geo_point or geo_shape) for the source',
+      async () => {
+        // Confirm point selected for default file, confirm shape also an option
+        let isGeoPointAvab = await PageObjects.maps.indexTypeOptionExists(GEO_POINT);
+        expect(isGeoPointAvab).to.be(true);
+        let isGeoShapeAvab = await PageObjects.maps.indexTypeOptionExists(GEO_SHAPE);
+        expect(isGeoShapeAvab).to.be(true);
+
+        // Upload shape file
+        const polygonJsonFile = 'polygon.json';
+        await PageObjects.maps.uploadJsonFileForIndexing(
+          path.join(__dirname, FILE_LOAD_DIR, polygonJsonFile)
+        );
+        await PageObjects.maps.waitForLayersToLoad();
+
+        // Confirm shape only option
+        isGeoPointAvab = await PageObjects.maps.indexTypeOptionExists(GEO_POINT);
+        expect(isGeoPointAvab).to.be(false);
+        isGeoShapeAvab = await PageObjects.maps.indexTypeOptionExists(GEO_SHAPE);
+        expect(isGeoShapeAvab).to.be(true);
+
+        // Multis
+        // Upload multipoint file
+        const multiPointJsonFile = 'multi_point.json';
+        await PageObjects.maps.uploadJsonFileForIndexing(
+          path.join(__dirname, FILE_LOAD_DIR, multiPointJsonFile)
+        );
+        await PageObjects.maps.waitForLayersToLoad();
+
+        // Confirm point selected for default file, confirm shape also an option
+        isGeoPointAvab = await PageObjects.maps.indexTypeOptionExists(GEO_POINT);
+        expect(isGeoPointAvab).to.be(true);
+        isGeoShapeAvab = await PageObjects.maps.indexTypeOptionExists(GEO_SHAPE);
+        expect(isGeoShapeAvab).to.be(true);
+
+        // Upload multipolygon file
+        const multiPolygonJsonFile = 'multi_polygon.json';
+        await PageObjects.maps.uploadJsonFileForIndexing(
+          path.join(__dirname, FILE_LOAD_DIR, multiPolygonJsonFile)
+        );
+        await PageObjects.maps.waitForLayersToLoad();
+
+        // Confirm point selected for default file, confirm shape also an option
+        isGeoPointAvab = await PageObjects.maps.indexTypeOptionExists(GEO_POINT);
+        expect(isGeoPointAvab).to.be(false);
+        isGeoShapeAvab = await PageObjects.maps.indexTypeOptionExists(GEO_SHAPE);
+        expect(isGeoShapeAvab).to.be(true);
+      });
+
+    it('should prevent import button from activating unless valid index name provided',
```

Comment:
Same here. This seems like a good test case for a jest test

Reply:
This I would prefer to leave in. There are a few pieces between file_upload and the map app that need to fall into place for this to happen